### PR TITLE
[lionkor-commandline] Add new port

### LINKED ISSUE #23459: [New Port Request] commandline
Library name: commandline

Library description: A C++ commandline for use in servers and chat software. Provides very simple asynchronous input/output.

Source repository URL: https://github.com/lionkor/commandline

### PR DESCRIPTION
**Pull request by Pospelove**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #23459 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JackBoosY** [comment]
When building x64-osx:
```
/Users/vagrant/Data/buildtrees/commandline/src/a4454e07a9-613e01ad85.clean/commandline.cpp:11:10: fatal error: 'conio.h' file not found
#include <conio.h>
         ^~~~~~~~~
1 error generated.
```

**Comment by Pospelove** [comment]
@JackBoosY marked osx as unsupported

**Comment by JackBoosY** [comment]
Waiting for test results.

File: ports/commandline/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "commandline",
+  "version-semver": "1.0.0",
+  "description": "A C++ commandline for use in servers and chat software. Provides very simple asynchronous input/output.",
+  "homepage": "https://github.com/lionkor/commandline/",
+  "license": "MIT",
+  "supports": "!osx & static",
```
Comment (JackBoosY) [inline ports/commandline/vcpkg.json:7]: Normally we don't use `static` here, we can add the following code to the top of `portfile.cmake`:
```cmake
vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```

File: ports/commandline/add-install.patch [hunk 2]
```diff
@@ -0,0 +1,9 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index b4db57c..7ace55c 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -22,3 +22,4 @@ if(BUILD_EXAMPLES)
+     target_link_libraries(test commandline)
+ endif()
+ 
++install(TARGETS commandline)
```
Comment (JackBoosY) [inline ports/commandline/add-install.patch:9]:
```suggestion
+install(
+  TARGETS commandline
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+)
```
Maybe it's better?

File: ports/commandline/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DBUILD_EXAMPLES=OFF
+)
+vcpkg_cmake_install()
+vcpkg_copy_pdbs()
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (JackBoosY) [inline ports/commandline/portfile.cmake:19]:
```suggestion
```
Install twice.

File: ports/commandline/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+# Handle header
+file(INSTALL "${SOURCE_PATH}/commandline.h" DESTINATION ${CURRENT_PACKAGES_DIR}/include)
```
Comment (JackBoosY) [inline ports/commandline/portfile.cmake:22]: Since you patched the source to install the target, I think you should also do this in the patch.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by Pospelove** [comment]
@JackBoosY Plz tell me if there is a way to disable `debug/include` via patch too

**Comment by ras0219-msft** [comment]
It's fine to just delete `debug/include` afterwards.

Additionally, the name of this library is extremely generic and there isn't a long history of associating that name with this library. Could you please rename the port to something more unique like `lionkor-commandline`?

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
See above comment.

**Comment by Pospelove** [comment]
@ras0219-msft No problem, I've just renamed everything via gh interface. hope it works :)

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**